tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4281,161 +4281,140 @@ TEST_F(ValidationTest, AtomicsInvalidDests) {
-    "Node node05 Thread launch type is not compatible with Compute");
-  RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node06 Thread launch type is not compatible with Compute");
+  if (m_ver.SkipDxilVersion(1, 7))
+    return;
+  std::vector<LPCWSTR> pArguments = {L"-HV", L"2021"};
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node07 Thread launch type is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node01' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node08 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node02' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node09 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node03' with input/outputs is not compatible with compute", false);
   RewriteAssemblyCheckMsg(
-    L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
-    pArguments.data(), 2, nullptr, 0,
-    {"!{i32 8, i32 15"},  // original: node shader
-    {"!{i32 8, i32 5"},   // changed to: compute shader
-    "Node node10 with input/outputs is not compatible with Compute");
+      L"..\\DXILValidation\\compute_node_compatibility.hlsl", "lib_6_8",
+      pArguments.data(), 2, nullptr, 0,
+      {"!{i32 8, i32 15"}, // original: node shader
+      {"!{i32 8, i32 5"},  // changed to: compute shader
+      "Node 'node04' coalescing launch type is not compatible with compute",
```

Comment:
When changing this to compute shader, shouldn't the error be different?  It's no longer identified as a node shader at all, so the error should really be that the metadata defines NodeLaunch metadata on a compute entry, right?